src/components/button/button.component.js

```diff
@@ -35,17 +96,20 @@ const renderStyledButton = (buttonProps) => {
 
   return (
     <StyledButton
+      as={!disabled && href ? "a" : "button"}
+      onKeyDown={href && handleLinkKeyDown}
+      draggable={false}
```

Comment:
Normally yes, but if you look at Martin's comment on this: https://github.com/Sage/carbon/issues/1756 - it shouldn't in this particular scenario because we want it to behave as a button, not a link